File: tldr/.local/bin/tldr/tldr.py
import sys
import subprocess

def fetch_cheat_sheet(query, full_output=False):
    try:
        # Construct the curl command
        command = f"curl -s cheat.sh/{query}"
        
        # Execute the curl command
        result = subprocess.run(command, shell=True, capture_output=True, text=True)
        
        # Check if the command was successful
        if result.returncode == 0:
            response = result.stdout
            if full_output:
                print(response)
            else:
                if "tldr:" in response:
                    print(response.split("tldr:")[1].strip())
                else:
                    print(response)
        else:
            print(f"Error: {result.stderr}")
    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python script.py [-f] <query>")
        sys.exit(1)

    full_output = False
    if sys.argv[1] == "-f":
        full_output = True
        query = sys.argv[2]
    elif len(sys.argv) > 2 and sys.argv[2] == "-f":
        full_output = True
        query = sys.argv[1]
    else:
        query = sys.argv[1]

    fetch_cheat_sheet(query, full_output)

File: tldr/.local/bin/tldr/tldr.rs
use std::env;
use std::process::Command;

fn fetch_cheat_sheet(query: &str, full_output: bool) {
    let output = Command::new("curl")
        .arg(format!("cheat.sh/{}", query))
        .output()
        .expect("Failed to execute curl");

    let response = String::from_utf8_lossy(&output.stdout);

    if full_output {
        println!("{}", response);
    } else {
        if let Some(tldr_start) = response.find("tldr:") {
            println!("{}", &response[tldr_start..]);
        } else {
            println!("{}", response);
        }
    }
}

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() < 2 {
        eprintln!("Usage: cargo run -- [-f] <query>");
        std::process::exit(1);
    }

    let mut full_output = false;
    let query: &str;

    if args[1] == "-f" {
        full_output = true;
        query = &args[2];
    } else if args.len() > 2 && args[2] == "-f" {
        full_output = true;
        query = &args[1];
    } else {
        query = &args[1];
    }

    fetch_cheat_sheet(query, full_output);
}

